### DISCUSSION TITLE
supabase joined tables filtering is not working

### DISCUSSION THREAD

**Discussion by gim96** [supabase/supabase#21358] [Questions] [answered]
table 1  (chat)  columns- id, type, name, profile_photo,is_favourite, updated_at
table 2 (chat_users) columns -  id, chat_id, user_id, is_read, is_mute (user_id is a foreign key in here)

I need to get above joined tables data by filtering user_id and type as follows:

 await supabase.from('chat')
    .select(
    `
      id, type, name, profile_photo,is_favourite, updated_at,
      chat_users!inner(id, user_id(id, raw_user_meta_data), is_read, is_mute)
    `)
    .match({ 'type':'group', 'chat_users.user_id.id': user.id })
    .then(async(resp) => {
      console.log('', resp.data)
     
    })
    .catch((err) => {
      console.log(err)
    })

But my result was like this: 
[
{
    "id": "f59063bc-853a-4d16-a6fd-a6023e5ff490",
    "type": "group",
    "name": "UNP",
    "profile_photo": "",
    "is_favourite": false,
    "updated_at": "2024-02-19T04:03:20.812559+00:00",
    "chat_users": [
        {
            "id": 330,
            "is_read": true,
            "is_mute": false,
            "user_id": null
        },
        {
            "id": 328,
            "is_read": true,
            "is_mute": false,
            "user_id": {
                "id": "3ae8228a-f9bd-4db0-b071-0951cd028235",
                "raw_user_meta_data": {
                    "name": "Gim Kelum",
                    "role": "Normal",
                    "color": "#800080",
                    "profile_photo": "",
                    "status": "available"
                }
            }
        },
        {
            "id": 329,
            "is_read": false,
            "is_mute": false,
            "user_id": null
        }
    ]
},
{
    "id": "824f8bba-3c14-48e2-baf0-a5f7a0666037",
    "type": "group",
    "name": "NEW",
    "profile_photo": "",
    "is_favourite": false,
    "updated_at": "2024-02-19T04:32:54.173175+00:00",
    "chat_users": [
        {
            "id": 335,
            "is_read": false,
            "is_mute": false,
            "user_id": null
        },
        {
            "id": 336,
            "is_read": true,
            "is_mute": false,
            "user_id": null
        },
        {
            "id": 334,
            "is_read": true,
            "is_mute": false,
            "user_id": null
        }
    ]
}
]

it returned all the data from db without filtering chat_users.user_id.id only filter chat.type.. how can i fix this ???

**Comment by GaryAustin1** [comment, marked as answer]
You could try `user_id!inner`.  I'm not sure that works, I could not find an example.
https://postgrest.org/en/v12/references/api/resource_embedding.html#null-filtering-on-embedded-resources also shows another filter that might be useful.

**Comment by gim96** [comment]
@GaryAustin1 it worked.! thank you!
